Euclidean_Space.R

```diff
@@ -0,0 +1,83 @@
+#Euclidean_Space
+
+library(reticulate)
+gs <- import_from_path("geomstats", path = ".")
+
+EuclideanSpace <- setRefClass("EuclideanSpace",
+                              contains = Manifold,
+                              methods = list(
+
+                                Init <- function(self, dimension){
+                                  stopifnot(dimension %% 1 == 0, dimension > 0)
+                                  self$dimension <- dimension
+                                  # self$metric <-
+                                },
+
+                                Belongs <- function(self,point){
+                                  # Evaluate if a point belongs to the Euclidean space.
+                                  point <- gs$backend$to_ndarray(point, to_ndim = 2)
+                                  n_points = point$shape
+                                  points_dim = point$shape
+                                  belongs = point_dim == self$dimension
+                                  belongs = gs$backend$repeat(belongs, repeats = n_points, axis = 0)
+                                  belongs = gs$backend$to_ndarray(belongs, to_ndim = 2, axis = 1)
+
+                                  return(belongs)
+                                },
+
+                                RandomUniform <- function(self, n_samples = 1){
+                                  #Sample in the Euclidean space with the uniform distribution.
+                                  size <- (n_samples, self$dimension)
+                                  point <- (gs$backend$random$rand(size) - 0.5) * 2
+                                  return(point)
+                                }
+
+                                )
+                              )
+
+
+EuclideanMetric <- setRefClass("EuclideanMetric",
+                               contains = gs$riemannian_metric,
+                               methods = list(
+
+                                 Init <- function(self, dimension){
+                                   stopifnot(dimension %% 1 == 0, dimension > 0)
+                                   #TODO(John): consider the super command
+                                 },
+
+                                 InnerProductMatrix <- function(self, base_point=None){
+                                   # Inner product matrix, independent of the base point.
+                                   
+                                   mat <- gs$backend$eye(self$dimension)
+                                   mat <- gs$backend$to_ndarray(mat, to_ndim = 3)

```

Comment:
Let's re-implement `to_ndarray` in R (see above comment). The python code for this is in geomstats/backend/numpy.y : see the `to_ndarray` function.